### ISSUE TITLE
[BUG]: "Limits start from top" incompatible with increased daily limit

### ISSUE THREAD

**Issue by virginviolet** [ankidroid/Anki-Android#15682]
### Checked for duplicates?

- [X] This issue is not a duplicate

### Bug

When the "limits start from top" option is enabled, increasing the daily limit with custom story or increasing the "today only" does not work. It just doesn't increase the number of new cards available. And yes, I'm not out of new cards, of course.

Or maybe it's supposed to be like that and I've misunderstood something.

### What are the steps to reproduce this bug?

Long tap a deck, select deck options.
Switch on "limits start from top".
Save.
Long tap any deck, select custom study, then "modify today's daily limit".
Enter any positive integer and tap OK.

### Expected behaviour

More new cards.

### Actual behaviour

No additional new cards.

### Debug info

```text
AnkiDroid Version = 2.17.1 (dd2f3762fb214f9dd51162b215974b23cc809713)

Backend Version = 0.1.34-anki23.12.1 (23.12.1 1a1d4d5419c6b57ef3baf99c9d2d9cf85d36ae0a)

Android Version = 14 (SDK 34)

ProductFlavor = full

Manufacturer = samsung

Model = SM-A346B

Hardware = mt6877

Webview User Agent = Mozilla/5.0 (Linux; Android 14; SM-XXXXX Build/UP1A.23XXXX.xxx; xx) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/122.0.6261.64 Mobile Safari/537.36

ACRA UUID = 73474d9c-c662-4436-afdd-f5404f157646

Crash Reports Enabled = true
```

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by virginviolet** [comment]
Oh... never mind. I had accidentaly(?) set the top deck to another deck category.

**Comment by virginviolet** [comment]
Buut... it seems likr you... sometimes? have to increase the limit on the top deck. I feel like that's s flaw. But I don't know, this thing is confusing the heck out of me, even though I want it!

**Comment by brishtibheja** [comment]
How is that a flaw? If limits start from the top and the parent deck has a limit of 10 the child deck cannot possibly have a limit of 15 right?

**Comment by virginviolet** [comment]
> How is that a flaw? If limits start from the top and the parent deck has a limit of 10 the child deck cannot possibly have a limit of 15 right?

I expected parent decks to have their temporary limit also increased if you increase a child deck. Partly because I could do that before I enabled "Limits start from the top", and it was very practical. But I guess you're right.

For some reason, this whole thing (limits) is very unintuitive and confusing for me. But I do prefer "limits start from the top", now that I've figured out how to work with it. I don't think I would have figured it out had I not made this issue.
